src/runtime_src/core/tools/common/tests/TestValidateUtilities.cpp

```diff
@@ -137,39 +137,26 @@ get_instr_size(const std::string& dpu_file) {
 
 void
 wait_for_max_clock(int& ipu_hclock, std::shared_ptr<xrt_core::device> dev) {
-  int ipu_hclock_pre = 0;
-  auto hclock_steady_counter = 0;
-  auto first_steady_state = -1, second_steady_state = -1;;
-
-  for(int i=0; i<100;i++){
+  uint64_t target_h_clock_freq = 0;
+  auto res_info = xrt_core::device_query<xrt_core::query::xrt_resource_raw>(dev);
```

Comment:
always use `device_query`. You never know which device doesn't support which query!

Reply:
Can you elaborate on this point ? I am using device_query only